homeassistant/components/hypontech/sensor.py

```diff
@@ -0,0 +1,109 @@
+"""The read-only sensors for Hypontech integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import HypontechConfigEntry, HypontechDataCoordinator
+from .hyponcloud import OverviewData
+
+
+@dataclass(frozen=True, kw_only=True)
+class HypontechSensorDescription(SensorEntityDescription):
+    """Describes Hypontech Inverter sensor entity."""
+
+    value_fn: Callable[[OverviewData], float | None]
+
+
+SENSORS: tuple[HypontechSensorDescription, ...] = (
+    HypontechSensorDescription(
+        key="pv_power",
+        translation_key="total_power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda c: c.power,
+    ),
+    HypontechSensorDescription(
+        key="lifetime_production",
+        translation_key="lifetime_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e_total,
+    ),
+    HypontechSensorDescription(
+        key="today_production",
+        translation_key="today_production",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda c: c.e_today,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HypontechConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+    coordinator = config_entry.runtime_data.coordinator
+
+    async_add_entities(
+        HypontechSensorWithDescription(coordinator, desc) for desc in SENSORS
+    )
+
+
+class HypontechSensorWithDescription(SensorEntity):
```

Comment:
Make this inherit a CoordinatorEntity, this can remove `available`, `async_added_to_hass` and `_handle_coordinator_update`